public/js/Filter.js

```diff
@@ -1,23 +1,26 @@
-function SampleFilter(uniqueRow, uniqueCol, uniquePlate, uniqueGene) {
+var _ = require('lodash');
+
+function SampleFilter(sampleData) {
```

Comment:
When you say docstrings, do you mean something that adds a ``__doc__`` type property that can get parsed by a documentation package? If so.. no, this is not included in JS natively (there's no way Netscape would've thought of that back in 1995 :P), but there are modules that will do this -- JSDoc comes to mind. Seeing as the UI isn't an API, I don't there's any point in those types of docstrings?

But if you just mean detailed comments at the beginning of each the modules and their methods, then yes that would probably worth doing!